### PR TITLE
Add the translations splitting script

### PR DESCRIPTION
**Pull request by Amr-Nash**
This script splits the combined translations from the I18N/I18N directory and distributes them to their modules

### CODE REVIEW

File: i18n_scripts/split_translation_files.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import os, localizable
+from collections import OrderedDict
+
+
+def separate_modules_translations(modules_dir, languages_dirs):
```
Comment (OmarIthawi) [inline i18n_scripts/split_translation_files.py]: docs please

File: i18n_scripts/split_translation_files.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    translations = {}
+    for lang_dir in languages_dirs:
+        file_path = os.path.join(modules_dir, 'I18N', 'I18N', lang_dir, 'Localizable.strings')
+        lang_list = localizable.parse_strings(filename=file_path)
+        for line in lang_list:
```
Comment (OmarIthawi) [inline i18n_scripts/split_translation_files.py]:
```suggestion
        for translation_entry in lang_list:
```

File: i18n_scripts/split_translation_files.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            module_name, _, key_remainder = line['key'].partition('.')
```
Comment (OmarIthawi) [inline i18n_scripts/split_translation_files.py]:
```suggestion
            module_name, _dot, key_remainder = line['key'].partition('.')
```

File: i18n_scripts/split_translation_files.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            module_name = module_name
```
Comment (OmarIthawi) [inline i18n_scripts/split_translation_files.py]:
```suggestion
```

File: i18n_scripts/split_translation_files.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            new_line = {
```
Comment (OmarIthawi) [inline i18n_scripts/split_translation_files.py]:
```suggestion
            split_entry = {
```

File: i18n_scripts/split_translation_files.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    return s
+
+
+def get_languages_dirs(modules_dir):
+    lang_parent_dir = os.path.join(modules_dir, 'I18N', 'I18N')
+    languages_dirs = [directory for directory in os.listdir(lang_parent_dir) if directory.endswith('.lproj')]
+    return languages_dirs
+
+
+def write_translations(modules_dir, modules_translations):
```
Comment (OmarIthawi) [inline i18n_scripts/split_translation_files.py]:
```suggestion
def write_translation_file(modules_dir, modules_translations):
```

File: i18n_scripts/split_translation_files.py [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        for lang_dir, translation_list in lang_dirs.items():
+            with open(os.path.join(modules_dir, module, module, lang_dir, 'Localizable.strings'), 'w') as f:
+                for line in translation_list:
+                    comment = line.get('comment', '')  # Retrieve the comment, if present
+                    if comment:
+                        f.write(f"/* {comment} */\n")
+                    f.write(f'"{line["key"]}" = "{(line["value"])}";\n')
+
+
+if __name__ == "__main__":
```
Comment (OmarIthawi) [inline i18n_scripts/split_translation_files.py]: convert into `def split_translation_files(modules_dir=None)`

File: i18n_scripts/split_translation_files.py [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    lang_parent_dir = os.path.join(modules_dir, 'I18N', 'I18N')
+    languages_dirs = [directory for directory in os.listdir(lang_parent_dir) if directory.endswith('.lproj')]
+    return languages_dirs
+
+
+def write_translations(modules_dir, modules_translations):
+    for module, lang_dirs in modules_translations.items():
+        for lang_dir, translation_list in lang_dirs.items():
+            with open(os.path.join(modules_dir, module, module, lang_dir, 'Localizable.strings'), 'w') as f:
+                for line in translation_list:
```
Comment (OmarIthawi) [inline i18n_scripts/split_translation_files.py]: line --> translation_entry

File: i18n_scripts/split_translation_files.py (hunk 1 above)
Comment (OmarIthawi) [reply i18n_scripts/split_translation_files.py]: also, please split into simpler funcitons such as:

```
def separate_module_translations(modules_dir, lang):
```


in main function we call it like the following:

```
for lang in langs:
    separate_modules_translations(modules_dir, lang)
```

File: i18n_scripts/split_translation_files.py [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+                    comment = line.get('comment', '')  # Retrieve the comment, if present
+                    if comment:
+                        f.write(f"/* {comment} */\n")
+                    f.write(f'"{line["key"]}" = "{(line["value"])}";\n')
+
+
+if __name__ == "__main__":
+    modules_dir = os.path.dirname(os.path.dirname(__file__))
+    languages_dirs = get_languages_dirs(modules_dir)
+    translations = separate_modules_translations(modules_dir, languages_dirs)
```
Comment (OmarIthawi) [inline i18n_scripts/split_translation_files.py]:
```suggestion
    for lang_dir in languages_dirs:
        translations = separate_modules_translations(modules_dir, lang_dir)
```

File: i18n_scripts/split_translation_files.py [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+        lang_list = localizable.parse_strings(filename=file_path)
+        for line in lang_list:
+            module_name, _, key_remainder = line['key'].partition('.')
+            module_name = module_name
+            new_line = {
+                'key': key_remainder,
+                'value': _escape(line['value']),
+                'comment': line['comment']
+            }
+            translations.setdefault(module_name, {}).setdefault(lang_dir, []).append(new_line)
```
Comment (OmarIthawi) [inline i18n_scripts/split_translation_files.py]: This line is rather complex, with the `for lang in langs` loop it should help to simplify it.

Comment (OmarIthawi) [reply i18n_scripts/split_translation_files.py]: This line is rather complex, with the `for lang in langs` loop it should help to simplify it:

```suggestion
            translations.setdefault(module_name, [])
            translations[module_name].append(split_entry)
```

File: i18n_scripts/split_translation_files.py [hunk 11]
```diff
@@ -0,0 +1,49 @@
+import os, localizable
+from collections import OrderedDict
+
+
+def separate_modules_translations(modules_dir, languages_dirs):
+    translations = {}
+    for lang_dir in languages_dirs:
```
Comment (OmarIthawi) [inline i18n_scripts/split_translation_files.py]: this one is moved outside

**Review by OmarIthawi — COMMENTED** [review]
Thanks a lot, it looks good. I've added many refactoring notes.

File: i18n_scripts/split_translation_files.py (hunk 8 above)
Comment (Amr-Nash) [reply i18n_scripts/split_translation_files.py]: done

File: i18n_scripts/split_translation_files.py (hunk 1 above)
Comment (Amr-Nash) [reply i18n_scripts/split_translation_files.py]: Done

File: i18n_scripts/split_translation_files.py (hunk 2 above)
Comment (Amr-Nash) [reply i18n_scripts/split_translation_files.py]: Dne

File: i18n_scripts/split_translation_files.py (hunk 3 above)
Comment (Amr-Nash) [reply i18n_scripts/split_translation_files.py]: Done

File: i18n_scripts/split_translation_files.py (hunk 4 above)
Comment (Amr-Nash) [reply i18n_scripts/split_translation_files.py]: Done

File: i18n_scripts/split_translation_files.py (hunk 5 above)
Comment (Amr-Nash) [reply i18n_scripts/split_translation_files.py]: Done

File: i18n_scripts/split_translation_files.py (hunk 7 above)
Comment (Amr-Nash) [reply i18n_scripts/split_translation_files.py]: Done

File: i18n_scripts/split_translation_files.py (hunk 9 above)
Comment (Amr-Nash) [reply i18n_scripts/split_translation_files.py]: Semplified

File: i18n_scripts/split_translation_files.py (hunk 10 above)
Comment (Amr-Nash) [reply i18n_scripts/split_translation_files.py]: Semplified

File: i18n_scripts/split_translation_files.py (hunk 11 above)
Comment (Amr-Nash) [reply i18n_scripts/split_translation_files.py]: done

File: i18n_scripts/split_translation_files.py [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+                    ...
+                ],
+                ...
+            }
+    """
+    translations = {}
+    file_path = os.path.join(modules_dir, 'I18N', lang_dir, 'Localizable.strings')
+    lang_list = localizable.parse_strings(filename=file_path)
+    for translation_entry in lang_list:
+        module_name, _dot, key_remainder = translation_entry['key'].partition('.')
```
Comment (OmarIthawi) [inline i18n_scripts/split_translation_files.py:37]:
```suggestion
        module_name, key_remainder = translation_entry['key'].split('.', maxsplit=1)
```


```
$ python
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> 'omar.ali.othman'.split('.', maxsplit=1)
['omar', 'ali.othman']
```

File: i18n_scripts/split_translation_files.py [hunk 13]
```diff
@@ -0,0 +1,121 @@
+import os, localizable
```
Comment (OmarIthawi) [inline i18n_scripts/split_translation_files.py:1]:
```suggestion
"""
Write a high-level doc for this script.
"""


import os, localizable
```

**Review by OmarIthawi — COMMENTED** [review]
I think this is ready! Thanks @Amr-Nash!!

File: i18n_scripts/split_translation_files.py (hunk 12 above)
Comment (Amr-Nash) [reply i18n_scripts/split_translation_files.py:37]: done

File: i18n_scripts/split_translation_files.py (hunk 13 above)
Comment (Amr-Nash) [reply i18n_scripts/split_translation_files.py:1]: done

File: i18n_scripts/split_translation_files.py (hunk 12 above)
Comment (OmarIthawi) [reply i18n_scripts/split_translation_files.py:37]: @Amr-Nash please push the changes. I don't see it in the pull request.

File: i18n_scripts/split_translation_files.py (hunk 13 above)
Comment (OmarIthawi) [reply i18n_scripts/split_translation_files.py:1]: @Amr-Nash please push the changes. I don't see it in the pull request.

**Review by OmarIthawi — APPROVED** [review]
I'm merging as-is. Please ensure changes are pushed in a new PR.

**Pull request merged**